src/utils/transition/linear-interpolator.js

```diff
@@ -11,17 +13,34 @@ const VIEWPORT_TRANSITION_PROPS = ['longitude', 'latitude', 'zoom', 'bearing', '
 export default class LinearInterpolator extends TransitionInterpolator {
 
   /**
-   * @param {Array} transitionProps - list of props to apply linear transition to.
+   * @param {Array} [transitionProps] - list of props to apply linear transition to.
+   * @param {Array} [opts] - additional options.
    */
-  constructor(transitionProps = VIEWPORT_TRANSITION_PROPS) {
+  constructor(transitionProps, opts) {
     super();
-    this.propNames = transitionProps;
+
+    if (Array.isArray(transitionProps)) {
```

Comment:
Feels hacky to differentiate between one arg vs two args, can we make constructor take an object? If this leads to a bigger change leave a comment here.

Reply:
This fix needs to go in as a patch. We can change the API in the next major release.

Reply:
Though I can make it take only one argument -- either an array (old) or an object (new)